code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -1233,6 +1233,20 @@
 		light(user,I)
 		update_brightness(user)
 		return
+	else if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/donut))
```

Comment:
Snack-level seems appropriate, though there should be a var or flag for whether an particular snack item can be dipped into things.